syntax = "proto2";

package userService;


service user_service {
  rpc AddStockToUser(stock_to_user_request) returns (ok_answer) {}
  rpc RemoveStockFromUser(stock_to_user_request) returns (ok_answer) {}
  rpc GetStocks(get_user_stocks_request) returns (stock_answer) {}
  rpc GetAllStocks(get_all_stocks_request) returns (stock_answer) {}
}


message stock_to_user_request {
  required string user = 1;
  required string stock_code = 2;
}

message get_user_stocks_request {
  required string user = 1;
}

message get_all_stocks_request {
}

message ok_answer {
  required string ok_string = 1;
}

message stock_answer {
  repeated string codes = 1;
}